accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+Overrides:
+
+- User can override the need to trust a repo by setting `DOTNET_USE_LOCAL` to true.
+- If `DOTNET_USE_LOCAL=true` and `.dotnet` directory is not found, it is not an error.
+- User can override using a local `.dotnet` directory (even if `global.json` requires it) by setting `DOTNET_USE_LOCAL=false` or by passing `--ignore-local` as an argument to `dotnet`.
+
+Notes:
+
+- This experience as written doesn't define or validate the behavior in Visual Studio. That needs more consideration. In theory, it should be transparent to Visual Studio.
+- Apphost will not honor this value, nor does it consult `global.json` in any scenario. Same with the `dotnet myapp.dll` pattern.
+
+Considerations for the future:
+
+- We could add a `dotnet install` command to enable private acqusition.
+- We could install .NET (local) Tools into the repo-resident `.dotnet` directory instead of into the user profile location. That approach might avoid the need for a local tools manifest.
+- It would be straightforward to build a version manager (like [DNVM](https://www.codeproject.com/Articles/1005145/DNVM-DNX-and-DNU-Understanding-the-ASP-NET-Runtime)) using symlinks for the `.dotnet` directory. This would be particularly useful for preview and nightly builds.
+
+## SDK version configuration
+
+SDK version selection is (currently) exclusively controlled via `global.json`. This is a file that can be placed anywhere on your machine (although typically within a repo). The `dotnet` host will look for it in the current directory or one of its (grand-)parents, including up to the root of the drive.
+
+```json
+{
+  "sdk": {
+    "version": "6.0.201",
+    "rollForward": "latestMajor",
+    "allowPrerelease": false
+  }
+}
+```
+
+This `global.json` example requires the 6.0.201 SDK or higher, and won't accept using preview versions (like .NET 7 Preview 1). There are other `global.json` configuration options, however, the three listed are the relevant ones.
+
+### Default SDK roll-forward behavior
+
+> Customer value: This feature satisfies the "intuitive version selection" and "Convenient and consistent infra builds"customer scenarios.
+
+By default, SDK version selection does not roll-forward past a feature band. For example, the following `global.json` file requires the use of .NET 7.0.100 and will accept any version through 7.0.199. It will not allow for the use of 7.0.200 (if 7.0.1xx builds are absent).
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+  }
+}
+```
+
+The feature bands have a reason to exist. With each feature band (`.200`, `.300`, ...), there is typically a significant update in a tools component like Roslyn, MSBuild, or NuGet. There may be compatibility breaks. However, if there are significant breaks, that's probably something that needs to resolve. The feature bands are not intended for intentional breaking changes.
```

Comment:
The SDK doesn't include intentional breaking changes in feature bands but tools have no awareness of feature bands. They treat them as VS releases and hold to no requirement of their being no breaking changes. We've been hit by this in the past where we thought the feature band was low risk but it turned out a change in nuget or roslyn was more significant. We may want to consider how to reconcile this with other tools teams as we can't limit them to one breaking change release a year but also don't want every feature band release to include customer risk.

Reply:
I feel like your comment and the text are saying the same thing. No?